internal/worker.go

```diff
@@ -205,6 +205,17 @@ type (
 		// Optional: Sets opentracing Tracer that is to be used to emit tracing information
 		// default: no tracer - opentracing.NoopTracer
 		Tracer opentracing.Tracer
+
+		// Optional: Enable worker for running shadowing workflows to replay existing workflows
+		// If set to true, worker will run in shadow mode and all other workers (decision, activity, session)
+		// will be disabled to prevent them from updating existing workflow states.
+		// default: false
+		EnableShadowWorker bool
```

Comment:
Could we use *ShadowOptions to indicate if the shadow worker is enabled?

Reply:
Yes, technically it will work. But I kind of think having a separate option for switching shadower on and off is more useful. So user can always have the same code for specifying the options, and using another option to turn it on or off. What's your thought?

Reply:
No strong opinion on it

Reply:
Second thought. Customers can comment out the shadow options if they temporary disable the shadow worker?